### PR TITLE
Support AFNI NIML surface data without node indices

### PR DESCRIPTION
**Pull request by nno**
Addresses #332.

### CODE REVIEW

File: mvpa2/support/nibabel/afni_niml.py [hunk 1]
```diff
@@ -285,12 +288,33 @@ def rawniml2string(p, form='text'):
         # remove some unncessary fields
         for f in ['vec_typ', 'vec_len', 'vec_num']:
             q.pop(f, None)
+    else:
+        has_body=False
```
Comment (yarikoptic) [inline mvpa2/support/nibabel/afni_niml.py]: PEP8: spaces around " = " in assignments

File: mvpa2/support/nibabel/afni_niml.py [hunk 2]
```diff
@@ -285,12 +288,33 @@ def rawniml2string(p, form='text'):
...
-    d = map(lambda x:x.encode(), ['<', '\n', ' >', '</', '>'])
-    return b''.join((d[0], s_name, d[1], s_header, d[2], s_body, d[3], s_name, d[4]))
+    if has_body:
+        delim = ['<', '\n', ' >', '</', '>']
+        values = [s_name, s_header, s_body, s_name]
+    else:
+        delim = ['<', '\n', '/>']
+        values = [s_name, s_header]
+
+    delim_enc=map(lambda x:x.encode(), delim)
```
Comment (yarikoptic) [inline mvpa2/support/nibabel/afni_niml.py]: PEP8 spaces here and below ;)  Try  pycharm editor -- it highlights all those nicely

File: mvpa2/tests/test_surfing_afni.py [hunk 3]
```diff
@@ -119,13 +120,37 @@ def _eq(p, q):
 
             for k in important_keys:
                 s_bad = s.replace(k.encode(), b'foo')
-                assert_raises((KeyError, ValueError), afni_niml.string2rawniml, s_bad)
+                #assert_raises((KeyError, ValueError), afni_niml.string2rawniml, s_bad)
```
Comment (yarikoptic) [inline mvpa2/tests/test_surfing_afni.py]: what will happen now?  what is the point of the loop then if nothing is checked -- smoke test?

File: mvpa2/tests/test_surfing_afni.py [hunk 4]
```diff
@@ -119,13 +120,37 @@ def _eq(p, q):
...
+                                d_empty_nodes[0]['nodes'][1].keys())
+
+
+
+
+
+
+
+
+
```
Comment (yarikoptic) [inline mvpa2/tests/test_surfing_afni.py]: that is way too generous! ;)

**Comment by yarikoptic** [comment]
I will merge but I would appreciate if you push directly those stylistic fixups

**Comment by nno** [comment]
Thanks @yarikoptic, I will work on these stylistic mixups and then push directly.

**Pull request merged**